pkg/plugins/policies/core/matchers/dataplane_test.go

```diff
@@ -206,4 +205,100 @@ var _ = Describe("MatchedPolicies", func() {
+			}),
+			Entry("k8s zone", dataplaneTestCase{
+				dataplaneMeta: test_resources.ZoneK8s,
+				policyMeta:    test_resources.ZoneK8s,
+				goldenFile:    buildGoldenFilePath("k8s-zone", givenResources.testName),
+			}),
+			Entry("policy global uni, dpp uni", dataplaneTestCase{
+				dataplaneMeta: test_resources.ZoneUni,
+				policyMeta:    test_resources.SystemPolicy(test_resources.GlobalUni),
+				goldenFile:    buildGoldenFilePath("policy-from-global-uni-zone-uni", givenResources.testName),
+			}),
+			Entry("policy global uni, dpp k8s", dataplaneTestCase{
+				dataplaneMeta: test_resources.ZoneK8s,
+				policyMeta:    test_resources.SystemPolicy(test_resources.GlobalUni),
+				goldenFile:    buildGoldenFilePath("policy-from-global-uni-zone-k8s", givenResources.testName),
+			}),
+			Entry("policy global k8s, dpp uni", dataplaneTestCase{
+				dataplaneMeta: test_resources.ZoneUni,
+				policyMeta:    test_resources.SystemPolicy(test_resources.GlobalK8s),
+				goldenFile:    buildGoldenFilePath("policy-from-global-k8s-zone-uni", givenResources.testName),
+			}),
+			Entry("policy global k8s, dpp k8s", dataplaneTestCase{
+				dataplaneMeta: test_resources.ZoneK8s,
+				policyMeta:    test_resources.SystemPolicy(test_resources.GlobalK8s),
+				goldenFile:    buildGoldenFilePath("policy-from-global-k8s-zone-k8s", givenResources.testName),
+			}),
+			Entry("policy global k8s, dpp uni", dataplaneTestCase{
+				dataplaneMeta: test_resources.ZoneUni,
+				policyMeta:    test_resources.SystemPolicy(test_resources.GlobalUni),
+				goldenFile:    buildGoldenFilePath("policy-global-uni-dpp-k8s", givenResources.testName),
+			}),
+			Entry("policy synced from other k8s zone", dataplaneTestCase{
+				dataplaneMeta: test_resources.ZoneUni,
+				policyMeta:    test_resources.ProducerPolicy(test_resources.SyncToUni(test_resources.ZoneK8s)),
+				goldenFile:    buildGoldenFilePath("policy-form-k8s-to-uni", givenResources.testName),
+			}),
+			Entry("policy synced from other k8s zone to k8s", dataplaneTestCase{
+				dataplaneMeta: test_resources.ZoneK8s,
+				policyMeta:    test_resources.ProducerPolicy(test_resources.SyncToK8s(test_resources.ZoneK8s)),
+				goldenFile:    buildGoldenFilePath("policy-form-k8s-to-k8s", givenResources.testName),
+			}),
+		)
+	}, generateTableEntries(filepath.Join("testdata", "matchedpolicies", "dataplane-kind")))
 })
+
+func getResourceType(resTypes []core_model.ResourceType) core_model.ResourceType {
+	var resType core_model.ResourceType
+	switch {
+	case len(resTypes) == 1:
+		resType = resTypes[0]
+	case len(resTypes) == 2 && resTypes[1] == v1alpha1.MeshHTTPRouteType:
+		resType = resTypes[0]
+	case len(resTypes) == 2 && resTypes[0] == v1alpha1.MeshHTTPRouteType:
+		resType = resTypes[1]
+	}
+	return resType
+}
+
+func buildGoldenFilePath(caseName, testName string) string {
+	return filepath.Join("testdata", "matchedpolicies", "dataplane-kind", testName, fmt.Sprintf("%s.golden.yaml", caseName))
```

Comment:
I'd put input files to the same `testName` directory as golden files

Reply:
it is not that simple and would require fighting with this method that generates this tests entries, I think we have more important stuff to focus on right now